runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/PuzzleSolverOverlay.java

```diff
@@ -230,23 +234,42 @@ else if (config.displayRemainingMoves())
 
 									PuzzleState futureMove = solver.getStep(j);
 
-									if (futureMove == null)
+									if (futureMove == null || !config.drawOverlaps() && positions.contains(futureMove.getEmptyPiece()))
 									{
 										break;
 									}
 
 									int blankX = futureMove.getEmptyPiece() % DIMENSION;
```

Comment:
I think this is better so the set isn't populated if we are drawing overlaps

```suggestion
									if (futureMove == null)
									{
										break;
									}
									
									if (!config.drawOverlaps())
									{
										if (positions.contains(futureMove.getEmptyPiece()))
										{
											break;
										}
										
										positions.add(futureMove.getEmptyPiece());
									}

									int blankX = futureMove.getEmptyPiece() % DIMENSION;
```